package NYT.NRpc.NProto;

import "yt/core/ytree/attributes.proto";
import "yt/core/misc/guid.proto";
import "yt/core/misc/error.proto";

////////////////////////////////////////////////////////////////////////////////

message TRequestHeader
{
    optional NYT.NProto.TGuid request_id = 1;
    required string path = 2;
    required string verb = 3;
    optional bool one_way = 4;
    optional NYT.NYTree.NProto.TAttributes attributes = 5;
    optional int64 request_start_time = 6;
    optional int64 retry_start_time = 7;

    extensions 100 to max;

}

////////////////////////////////////////////////////////////////////////////////

message TAuthenticatedExt
{
    extend NRpc.NProto.TRequestHeader
    {
        optional string authenticated_user = 102;
    }
}

message TResponseHeader
{
    optional NYT.NProto.TGuid request_id = 1;
    required NYT.NProto.TError error = 2;
    optional NYT.NYTree.NProto.TAttributes attributes = 3;
}

////////////////////////////////////////////////////////////////////////////////
